sql/hive/src/test/scala/org/apache/spark/sql/hive/client/HiveShimSuite.scala

```diff
@@ -0,0 +1,87 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.hive.client
+
+import org.apache.hadoop.hive.conf.HiveConf
+import org.apache.hadoop.hive.metastore.TableType
+import org.apache.hadoop.hive.ql.metadata.{Hive, Table}
+import org.apache.hadoop.hive.ql.session.SessionState
+
+import org.apache.spark.SparkUnsupportedOperationException
+import org.apache.spark.sql.hive.client.HiveClientImpl.getHive
+import org.apache.spark.util.Utils
+
+class HiveShimSuite(version: String) extends HiveVersionSuite(version) {
+
+  private val shim = version match {
+    case "0.12" => new Shim_v0_12()
+    case "0.13" => new Shim_v0_13()
+    case "0.14" => new Shim_v0_14()
+    case "1.0" => new Shim_v1_0()
+    case "1.1" => new Shim_v1_1()
+    case "1.2" => new Shim_v1_2()
```

Comment:
This is removed by https://github.com/apache/spark/commit/3c84c229d167a6ab2857649e91fff6f0d57bb12c
cc @HyukjinKwon

Reply:
Yeah we don't support them anymore after JDK upgrade.

Reply:
Done.
Deleted 0.12-1.2 versions and rebased to master.